ambry-store/src/main/java/com/github/ambry/store/CompactionManager.java

```diff
@@ -173,15 +178,30 @@ boolean removeBlobStore(BlobStore store) {
     return true;
   }
 
+  /**
+   * Get the current replicaToCounterMap policy.
+   * @return {@link this.replicaToCounterMap}
+   */
+  Map<ReplicaId, CompactionPolicyCounter> getReplicaToCounterMap(){
+    return this.replicaToCounterMap;
+  }
+
   /**
    * Get compaction details for a given {@link BlobStore} if any
    * @param blobStore the {@link BlobStore} for which compaction details are requested
    * @return the {@link CompactionDetails} containing the details about log segments that needs to be compacted.
    * {@code null} if compaction is not required
    * @throws StoreException when {@link BlobStore} is not started
    */
-  private CompactionDetails getCompactionDetails(BlobStore blobStore) throws StoreException {
-    return blobStore.getCompactionDetails(compactionPolicy);
+  CompactionDetails getCompactionDetails(BlobStore blobStore) throws StoreException {
+    ReplicaId replicaId = blobStore.getReplicaId();
+    if (compactionPolicyFactory != null && compactionPolicyFactory instanceof HybridCompactionPolicyFactory) {
+      CompactionPolicyCounter compactionPolicyCounter =
+          replicaToCounterMap.getOrDefault(replicaId, new CompactionPolicyCounter(storeConfig));
+      compactionPolicyCounter.increment();
+      replicaToCounterMap.put(replicaId, compactionPolicyCounter);
+    }
```

Comment:
we actually don't have to differentiate the compaction policy factory when creating and incrementing the couter. We can just increment it for every policy.

Reply:
Move map inside Hybrid compaction policy.